core/federated/RTI/message_record/message_record.c

```diff
@@ -0,0 +1,170 @@
+/**
+ * @file message_record.c
+ * @author Soroush Bateni (soroush@berkeley.edu)
+ * @brief Record-keeping for in-transit messages.
+ * @version 0.1
+ * @date 2022-06-02
+ * 
+ * @copyright Copyright (c) 2022, The University of California at Berkeley.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+#include "message_record.h"
+#include <stdlib.h>
+
+/**
+ * @brief Initialize the in-transit message record queue.
+ * 
+ * @return in_transit_message_record_q 
+ */
+in_transit_message_record_q_t initialize_in_transit_message_q() {
+    in_transit_message_record_q_t queue = 
```

Comment:
I don't think this should be a calloc.  You can just declare an in_transit_message_record_q_t on the stack and then return it. The returned value will be a copy of the stack variable, which is a pair of pointers.  In any case, this looks wrong because calloc always returns a pointer and here you cast it to a non-pointer, which will likely result in overwriting 64 bits of stack data.

Reply:
`in_transit_message_record_q_t` is a pointer:
```C
typedef struct in_transit_message_record_q* in_transit_message_record_q_t;
```

Reply:
@edwardalee did my comments address your concern? This PR is still pending an approval.